doc/introduction/returns.rst

```diff
@@ -0,0 +1,170 @@
+
+    import pennylane as qml
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.expval(qml.PauliZ(0)), qml.probs(0)
+
+In version 0.29 and earlier of PennyLane, ``circuit()`` would return a single length-3 array:
+
+.. code-block:: pycon
+
+    >>> circuit(0.5)
+    tensor([0.87758256, 0.93879128, 0.06120872], requires_grad=True)
+
+In versions 0.30 and above, ``circuit()`` returns a length-2 tuple containing the expectation value
+and probabilities separately:
+
+.. code-block:: pycon
+
+    >>> circuit(0.5)
+    (tensor(0.87758256, requires_grad=True),
+     tensor([0.93879128, 0.06120872], requires_grad=True))
+
+For a detailed explanation of this change, along with code examples, check out the
+:func:`~.enable_return` function documentation.
+
+Motivation
+----------
+
+PennyLane has historically adopted the approach of combining the returned
+:ref:`measurements <intro_ref_meas>` of a QNode into a single array. However, this has presented
+some challenges:
+
+* The return of a QNode could be different from what is expected, as shown in the
+  :ref:`example <return-type-example-issue>` above.
+* For measurements of different shapes, ragged arrays were generated internally and then squeezed
+  into a single output array. This is incompatible with NumPy's
+  `NEP 34 <https://numpy.org/neps/nep-0034-infer-dtype-is-object.html>`_ and constrains the
+  `version of NumPy <https://github.com/PennyLaneAI/pennylane/blob/v0.29.1/setup.py#L21>`_ that
+  PennyLane is compatible with.
+* Use of stacking and squeezing presents performance bottlenecks.
+
+The changes made in version 0.30 of PennyLane address the challenges above. However, existing users
+may experience breaking changes or issues when upgrading their PennyLane version.
+
+.. _Troubleshooting:
+
+Troubleshooting
+---------------
+
+You may experience issues with PennyLane's updated QNode return system in version 0.30 and above
+if you have existing code that works with an earlier version of PennyLane. To help identify a fix,
+select the option below that describes your situation.
+
+.. details::
+    :title: I am using a QNode that returns more than one quantity
+    :href: qnode-multiple-returns
+
+    Your issue may be because:
+
+    * You are calculating the Jacobian of the QNode using the :ref:`NumPy <numpy_interf>` or
+      :ref:`TensorFlow <tf_interf>` interface. For example, the following will now raise an error:
+
+      .. code-block:: python
+
+          from pennylane import numpy as np
+
+          dev = qml.device("default.qubit", wires=1)
+
+          @qml.qnode(dev)
+          def circuit(x):
+              qml.RX(x, wires=0)
+              return qml.expval(qml.PauliY(0)), qml.expval(qml.PauliZ(0))
+
+          x = np.array(0.5, requires_grad=True)
+          qml.jacobian(circuit)(x)
+
+      Follow the instructions :ref:`here <return-autograd-tf-gotcha>` to fix this issue.

```

Comment:
Should we move those instructions to this page?

Reply:
Up for discussion, but I feel like that page covers:
- Details usage examples for interfaces
- Highlighting behaviour with shot vectors, batching
- Gotchas

Whereas this page is more for summarizing the change and helping people solve the problem.

Combining together feels like it might balloon the page. Moreover, I'm not sure if either page will stick around long term.

Reply:
It made sense to have all that information there because that's how people opted-in for that behavior.  Now that the behaviour is by default, I think it should be here instead. Most users shouldn't care about the `qml.enable_return` toggle.

Reply:
I've had a go, but I'm not sure I like the how the outcome looks:
![image](https://user-images.githubusercontent.com/49409390/234975132-6c5528b4-da4f-423b-8e6a-f52f502d7577.png)

Overall, I think that this doc should be kept lightweight, focusing on the high-level implications and troubleshooting, rather than the details.

Reply:
One thing I have done though is to remove the part in `enable_return` that says this is experimental: 960157bdbfc3f5ff31e0bb64aa4c11257f5422b7